tests/equilibria/test_constraint_functions.py

```diff
@@ -0,0 +1,159 @@
+    CoilForceConstraint,
+    CurrentMidplanceConstraint,
+    FieldConstraintFunction,
+    L2NormConstraint,
+)
+
+TEST_PATH = get_bluemira_path("equilibria/test_data", subfolder="tests")
+
+# TODO add control coil test
+
+
+class TestSimpleABConstraintFuntions:
+    @classmethod
+    def setup_class(cls):
+        cls.vector = np.array([2, 2])
+        cls.a_mat = np.array([[1, 0], [0, 1]])
+        cls.b_vec = np.array([1, 1])
+        cls.value = [2.0, 0.2]
+        cls.scale = [1.0, 0.1]
+
+    def constrint_setup(self, v, s, constraint):
+        con_setup = constraint(
+            a_mat=self.a_mat,
+            b_vec=self.b_vec,
+            value=v,
+            scale=s,
+        )
+        print(v, s)
+        return con_setup
+
+    def test_AxBConstraint(self):
+        f_res = [-1.0, -1.0]
+        df_res = [1.0, 0.1]
+        for v, s, f, df in zip(self.value, self.scale, f_res, df_res):
+            axb = self.constrint_setup(v, s, AxBConstraint)
+            test_f = axb.f_constraint(self.vector)
+            test_df = axb.df_constraint(self.vector)
+            assert isinstance(test_f, np.ndarray)
+            assert isinstance(test_df, np.ndarray)
+            assert all(test_f == f)
+            assert all(np.diag(test_df) == df)
+
+    def test_L2NormConstraint(self):
+        f_res = [0.0, 1.08]
+        # df_res = [2.0, -0.16]
+        for v, s, f in zip(self.value, self.scale, f_res):
+            axb = self.constrint_setup(v, s, L2NormConstraint)
+            test_f = axb.f_constraint(self.vector)
+            test_df = axb.df_constraint(self.vector)
+            assert isinstance(test_f, float)
+            assert isinstance(test_df, np.ndarray)
+            assert test_f == pytest.approx(f)
+
+
+class TestEquilibriumInput:
+    @classmethod
+    def setup_class(cls):
+        eq_name = "eqref_OOB.json"
+        cls.eq = Equilibrium.from_eqdsk(Path(TEST_PATH, eq_name))
+        cls.coilset = cls.eq.coilset
+        cls.vector = cls.coilset.current
+        cls.scale = 1.0
+
+    def test_feild_constraint_function(self):
+        x_test = [4.0, 4.0]
+        z_test = [8.0, 6.0]
+        B_max = [5.0, 5.0]
+        const = [-2.0, -2.0]
+        for x, z, b, c in zip(x_test, z_test, B_max, const):
+            ax_mat = self.coilset.Bx_response(x, z, control=True)
+            az_mat = self.coilset.Bz_response(x, z, control=True)
+            bxp_vec = np.atleast_1d(self.eq.Bx(x, z))
+            bzp_vec = np.atleast_1d(self.eq.Bz(x, z))
+            fcf = FieldConstraintFunction(
+                ax_mat=ax_mat,
+                az_mat=az_mat,
+                bxp_vec=bxp_vec,
+                bzp_vec=bzp_vec,
+                B_max=b,
+                scale=self.scale,
+            )
+
+            # TODO check df result
+            assert len(fcf.f_constraint(self.vector)) == 1
+            assert fcf.f_constraint(self.vector) == pytest.approx(c, 0.1)
+
+    def test_current_midplance_constraint(self):
```

Comment:
```suggestion
    def test_current_midplane_constraint(self):
```

I realise that `CurrentMidplanceConstraint` also has the same spelling mistake in it.

Maybe that should be changed too?